grakn-bootup/src/main/java/ai/grakn/bootup/graknengine/pid/UnixGraknPidRetriever.java

```diff
@@ -32,36 +29,13 @@
  */
 
 public class UnixGraknPidRetriever implements GraknPidRetriever {
-    public static final String psEfCommand = "ps -ef | ps -ef | grep \"ai.grakn.bootup.graknengine.Grakn\" | grep -v grep | awk '{print $2}'";
     public long getPid() {
-        StringBuilder outputS = new StringBuilder();
-        int exitValue = 1;
-
-        Process p;
-        try {
-            p = Runtime.getRuntime().exec(new String[] { "/bin/sh", "-c", psEfCommand }, null, null);
-            p.waitFor();
-            exitValue = p.exitValue();
-
-            if (exitValue == 0) {
-                try (BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream(), StandardCharsets.UTF_8))){
-                    String line;
-                    while ((line = reader.readLine()) != null) {
-                        outputS.append(line).append("\n");
-                    }
-                }
-            } else {
-                throw new RuntimeException("a non-zero exit code '" + exitValue + "'returned by the command '" + psEfCommand + "'");
-            }
-        } catch (InterruptedException | IOException e) {
-            // DO NOTHING
-        }
-
-        String pidString = outputS.toString().trim();
+        String[] pidAndHostnameString = ManagementFactory.getRuntimeMXBean().getName().split("@");
+        String pidString = pidAndHostnameString[0];
```

Comment:
Do we know how portability is affected? isn't the name platform-specific? That's what the `RuntimeMXBean` spec says.

Reply:
@kasper-piskorski There exists no platform-independent way that can be guaranteed to work in all jvm implementations. ManagementFactory.getRuntimeMXBean().getName() looks like the best (closest) solution. It's short, and probably works in every implementation in wide use.

On linux+windows it returns a value like 12345@hostname (12345 being the process id).